README.md

```diff
@@ -0,0 +1,400 @@
+It should be convenient to start each functionality block with a comment.
+
+### Horizontal Whitespace
+
+There should not be trailing whitespace at the end of the line.
+
+#### General
+
+When declaring variables, there must be only one space between type and variable name.
+If several variables are declared in the same line, there must be a space after the comma, but not before.
+This also applies in function declarations and callings.
+
+There must be no space before semicolons.
+
+When declaring arrays, the opening square bracket `[` should be always adjacent to the variable name.
+
+Before a backslash-newline `\` there should be a space.
+
+In `case` statements, there should not be a space before the colon `:`.
+
+#### Loops and Conditionals
+
+There must be a space between the keyword (`if`, `for`, `switch`, `while`, etc.) and the open parenthesis `(`.
+There should not be a space after the closing parenthesis `)` of a control statement.
+
+In `for` loops, there is always one space after each semicolon.
+
+#### Operators
+
+Assignment operators always have spaces around them.
+This also affects to compound assignment operators.
+
+Other binary arithmetic operators must also have spaces around them.
+This does not apply to the sign and the increment `++` and decrement `--`, which should be adjacent to the variable it affects.
+
+```c++
+v = -5 * x + y / --z;
+
+x += y++;
+```
+
+Logical operators `&&` and `||` should also have spaces around them.
+This does not apply to unary logical operator `!` which should be written adjacent to the variable it affects.
+
+Similarly, bitwise operators should have spaces around them, except the unary complement (bit inversion) operator `~` which should also be written adjacent to the variable it affects.
+
+Comparison operators ought to have spaces around them too.
+
+```c++
+if (param1 == 0 && param1 > 3 || !param2)
+{
+}
+```
+
+There should be no space around member operators `.` and `->`.
+
+Address-of `&` and de-reference `*` unary operators should be always adjacent to the variable they affect.
+
+When overloading an operator, there should be a space between the `operator` keyword and the operator being overloaded.
+
+```c++
+Subscriber& Subscriber::operator <<(
+        const qos::SubscriberQos& qos)
+{
+    this->qos(qos);
+    return *this;
+}
+```
+
+#### Templates
+
+There should be a space between closing angles `>`.
+
+```c++
+std::map<std::string, std::vector<DataWriterImpl*> > writers_;
+```
+
+#### Line-end Comments
+
+Lines that are non-obvious should get a comment at the end of the line.
+These end-of-line comments should be separated from the code by a space.
+
+### Function Declarations and Definitions
+
+To improve parameter readability, each parameter must be in its own line, even the first one and even if there is only one parameter.
+The line-wrapping indentation is to be used in this case.
+Never indent to the opening parenthesis level, the first parameter, etc.
```

Comment:
The first parameter, if the rules are followed, should be on its own line and indented with 8 spaces, just as the remaining parameters. This wording may cause some confusion. I'd simplify it for the sake of clarity.
```suggestion
Never indent to the opening parenthesis level.
```